pythonforandroid/recipe.py

```diff
@@ -577,12 +581,15 @@ def list_recipes(cls, ctx):
                         yield name
 
     @classmethod
-    def get_recipe(cls, name, ctx):
+    def get_recipe(cls, name, ctx, build_as_debuggable=False):
         '''Returns the Recipe with the given name, if it exists.'''
         name = name.lower()
         if not hasattr(cls, "recipes"):
             cls.recipes = {}
         if name in cls.recipes:
+            if ctx is not None:
+                cls.recipes[name].ctx = ctx
+            cls.recipes[name].build_as_debuggable = build_as_debuggable
```

Comment:
This is really really confusing me. Why do we even have to do this since it's been done here below via:
```python
recipe.build_as_debuggable = build_as_debuggable
recipe.ctx = ctx
cls.recipes[name.lower()] = recipe
```
Plus the whole thing seems unnatural to me, we're passing Recipe attributes from here while it's usually set a recipe level, feels like we're missing design patterns/vision somehow, no?
Couldn't the context be used to pass such things around?

Reply:
@AndreMiras for what it's worth the additional `cls.recipes[name].ctx = ctx` here is vital because the unit tests will call this function and obtain the same recipe via *multiple* contexts (for the different tests) and break otherwise in weird ways

Regarding `build_as_debuggable`, you are completely right. I will look into moving it :+1: :clap: makes more sense on the context

Reply:
Thanks for looking into this :hearts: